### ISSUE TITLE
JSON serialization bug with `if_`

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-python-client#470]
issue reported by @mbuchhorn

minimal reproduction snippet:

```python
import openeo
from openeo.processes import if_

cube = connection.load_collection("CGLS_GDMP300_V1_GLOBAL")
cube = if_(True, cube, cube)
cube = cube.apply_dimension(process=openeo.UDF(code="def x(): pass"), dimension='t')

cube.print_json()
# - > TypeError: Object of type UDF is not JSON serializable
```

Dropping the `if_` line makes it work again

**Comment by soxofaan** [comment]
workaround with current version of python client is to replace the `cube = if_(True, cube, cube)` with
```
cube = cube.process(process_id="if", arguments={"value": True, "accept": cube, "reject": cube})
```

**Comment by mbuchhorn** [comment]
This UDP with same _if and UDF call works: https://github.com/integratedmodelling/OpenEO-UDP-UDF-catalogue/blob/main/UDP/UDP_CORINE_arable_mask.py

only difference: "apply_dimension" instead of apply in the not working version

**Comment by soxofaan** [comment]
@mbuchhorn  no it's the other `if_` in your original failing code

this (with apply iso apply_dimension) also fails

```
cube = connection.load_collection("CGLS_GDMP300_V1_GLOBAL")
cube = if_(True, cube, cube)
cube = cube.apply(process=openeo.UDF(code="def x(): pass"))
```

**Comment by soxofaan** [comment]
fixed (will be included in next release 0.23.0)
